packages/medusa/src/api/admin/product-options/[id]/route.ts

```diff
@@ -0,0 +1,71 @@
+import {
+  deleteProductOptionsWorkflow,
+  updateProductOptionsWorkflow,
+} from "@medusajs/core-flows"
+import {
+  AuthenticatedMedusaRequest,
+  MedusaResponse,
+} from "@medusajs/framework/http"
+
+import {
+  AdminGetProductOptionParamsType,
+  AdminUpdateProductOptionType,
+} from "../validators"
+import { HttpTypes } from "@medusajs/framework/types"
+import { ContainerRegistrationKeys } from "@medusajs/framework/utils"
+
+export const GET = async (
+  req: AuthenticatedMedusaRequest<AdminGetProductOptionParamsType>,
+  res: MedusaResponse<HttpTypes.AdminProductOptionResponse>
+) => {
+  const query = req.scope.resolve(ContainerRegistrationKeys.QUERY)
+  const {
+    data: [product_option],
+  } = await query.graph({
+    entity: "product_option",
+    filters: { id: req.params.id },
+    fields: req.queryConfig.fields,
+  })
+
+  res.status(200).json({ product_option })
+}
+
+export const POST = async (
+  req: AuthenticatedMedusaRequest<AdminUpdateProductOptionType>,
+  res: MedusaResponse<HttpTypes.AdminProductOptionResponse>
+) => {
+  const { result } = await updateProductOptionsWorkflow(req.scope).run({
+    input: {
+      selector: { id: req.params.id },
+      update: req.validatedBody,
+    },
+  })
+
+  const query = req.scope.resolve(ContainerRegistrationKeys.QUERY)
+  const {
+    data: [product_option],
+  } = await query.graph({
+    entity: "product_option",
+    filters: { id: result[0].id },
+    fields: req.queryConfig.fields,
+  })
+
+  res.status(200).json({ product_option })
+}
+
+export const DELETE = async (
+  req: AuthenticatedMedusaRequest,
+  res: MedusaResponse<HttpTypes.AdminProductOptionDeleteResponse>
+) => {
+  const id = req.params.id
+
+  await deleteProductOptionsWorkflow(req.scope).run({
```

Comment:
q: do we validate if a global option can be deleted if there are linked products?